cookbooks/custom_formatter.rst

```diff
@@ -0,0 +1,465 @@
+Writing a custom Behat formatter
+================================
+
+How to write a custom formatter for Behat?
+
+Introdution
+-----------
+
+Why a custom formatter?
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Behat has three native formatters:
+
+-  **pretty**: the default formatter, which prints every line in green (if a test passes) or red (if it fails),
+-  **progress**: print a "dot" for each test, and a recap of all failing tests at the end,
+-  **junit**: outputs a `junit <https://junit.org/>`__ compatible XML file.
+
+Those are nice, and worked for most of the cases. You can use the "progress" one for the CI, and the "pretty" for development for example.
+
+But you might want to handle differently the output that Behat renders.
+In this cookbook, we will see how to implement a custom formatter for `reviewdog <https://github.com/reviewdog/reviewdog>`__,
+a global review tool that takes input of linters or testers, and that can send "checks" on github, bitbucket or gitlab PR.
+
+Reviewdog can handle `two types of input <https://github.com/reviewdog/reviewdog#input-format>`__:
+
+-  any stdin, coupled with an "errorformat" (a Vim inspired format that can convert text string to machine-readable errors),
+-  a `"Reviewdog Diagnostic Format" <https://github.com/reviewdog/reviewdog/tree/48b25a0aafb8494e751387e16f729faee9522c46/proto/rdf>`__: a JSON with error data that reviewdog can parse.
+
+In my case, I tried parsing Behat's output with errorformat, but I do not know this language, and the multi-line Behat output with "dots" didn't make it easy.
```

Comment:
I don't think the documentation should by using `I` here. this looks fine for a blog post (where `I` refers to the author of the post), but not so much for documentation.